### ISSUE TITLE
hostapd: 2.10 breaks ap functionality

### ISSUE THREAD

**Issue by zlice** [void-linux/void-packages#35420]
### System

* xuname:  
`Void 5.15.12_1 x86_64 AuthenticAMD notuptodate rFF`

* package:  
`hostapd-2.9_3`

### Expected behavior
Wireless AP starts

### Actual behavior

*dmesg*
```
[  943.200854] br0: port 5(wifi0) entered blocking state
[  943.200861] br0: port 5(wifi0) entered disabled state
[  943.200957] device wifi0 entered promiscuous mode
[  943.201616] device wifi0 left promiscuous mode
[  943.201650] br0: port 5(wifi0) entered disabled state
```

*hostapd*
```
hostapd -S /etc/hostapd/hostapd.conf
ACS: Automatic channel selection started, this may take a bit
ACS: No available channels found
wifi0: IEEE 802.11 Configured channel (0) or frequency (0) (secondary_channel=0) not found from the channel list of the current mode (2) IEEE 802.11a
wifi0: IEEE 802.11 Hardware does not support configured channel
Could not select hw_mode and channel. (-3)
wifi0: interface state UNINITIALIZED->DISABLED
wifi0: AP-DISABLED 
wifi0: Unable to setup interface.
wifi0: interface state DISABLED->DISABLED
wifi0: AP-DISABLED 
wifi0: CTRL-EVENT-TERMINATING 
hostapd_free_hapd_data: Interface wifi0 wasn't started
nl80211: deinit ifname=wifi0 disabled_11b_rates=0
nl80211: Failed to remove interface wifi0 from bridge br0: Invalid argument
```

### Steps to reproduce the behavior

*lspci*
`0b:00.0 Network controller: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter (rev 32)`

tried changing channels, `iw reg set` region, moving around bridge and interfaces, "wds" and "4addr" settings i found. reverting to 2.9 allows AP to start.

**Comment by zlice** [comment]
more precise `-dd`

```
Allowed channel: mode=0 chan=10 freq=2457 MHz max_tx_power=30 dBm
Allowed channel: mode=0 chan=11 freq=2462 MHz max_tx_power=30 dBm
ACS: Automatic channel selection started, this may take a bit
ACS: No available channels found
wifi0: IEEE 802.11 Configured channel (0) or frequency (0) (secondary_channel=1) not found from the channel list of the current mode (2) IEEE 802.11a
wifi0: IEEE 802.11 Hardware does not support configured channel
Could not select hw_mode and channel. (-3)
wifi0: interface state UNINITIALIZED->DISABLED
wifi0: AP-DISABLED
```

Pretty sure this is a hostapd, firmware or kernel bug (or a combination or all three? i can't keep up with wifi on linux, swear something changes every time a dev sneezes thinking of wifi). But I have tried everything at this point, hostapd-2.10 is broken.

**Comment by zlice** [comment]
Found a uh, workaround. Hardware works mostly, hostapd "works" mostly. So I have to assume it's a "fix" in 2.10 and firmware that's labeling things as not-usable when they damn well are and should be. lil mad. closing.
